modules/core/src/nan_mask.simd.hpp

```diff
@@ -0,0 +1,280 @@
+    val.f = (float)newVal;
+
+    size_t j = 0;
+
+#if (CV_SIMD || CV_SIMD_SCALABLE)
+    v_int32 v_mask1 = vx_setall_s32(0x7fffffff), v_mask2 = vx_setall_s32(0x7f800000);
+    v_int32 v_val = vx_setall_s32(val.i);
+
+    size_t cWidth = (size_t)VTraits<v_int32>::vlanes();
+    for (; j + cWidth <= len; j += cWidth)
+    {
+        v_int32 v_src = vx_load(tptr + j);
+        v_int32 v_cmp_mask = v_lt(v_mask2, v_and(v_src, v_mask1));
+        v_int32 v_dst = v_select(v_cmp_mask, v_val, v_src);
+        v_store(tptr + j, v_dst);
+    }
+    vx_cleanup();
+#endif
+
+    for (; j < len; j++)
+    {
+        if ((tptr[j] & 0x7fffffff) > 0x7f800000)
+        {
+            tptr[j] = val.i;
+        }
+    }
+}
+
+
+static void patchNaNs_64f(uchar* ptr, size_t len, double newVal)
+{
+    int64_t* tptr = (int64_t*)ptr;
+    Cv64suf val;
+    val.f = newVal;
+
+    size_t j = 0;
+
+#if (CV_SIMD || CV_SIMD_SCALABLE)
+    v_int64 v_mnt_mask = vx_setall_s64(0x000FFFFFFFFFFFFF);
+    v_int64 v_exp_mask = vx_setall_s64(0x7FF0000000000000);
+    v_int64 v_val = vx_setall_s64(val.i);
+
+    size_t cWidth = (size_t)VTraits<v_int64>::vlanes();
+    for (; j + cWidth <= len; j += cWidth)
+    {
+        v_int64 v_src = vx_load(tptr + j);
+        v_int64 vande = v_and(v_src, v_exp_mask);
+        v_int64 vandm = v_and(v_src, v_mnt_mask);
+        v_int64 ve, vm;
+
+        ve = v_eq(vande, v_exp_mask);
+        vm = v_ne(vandm, vx_setzero_s64());
+
+        v_int64 v_isnan = v_and(ve, vm);
+        v_int64 v_dst = v_or(v_and(v_isnan, v_val), v_and(v_not(v_isnan), v_src));
+        v_store(tptr + j, v_dst);
+    }
+    vx_cleanup();
+#endif
+
+    for (; j < len; j++)
+        if ((tptr[j] & 0x7FFFFFFFFFFFFFFF) > 0x7FF0000000000000)
+            tptr[j] = val.i;
+}
+
+PatchNanFunc getPatchNanFunc(bool isDouble)
+{
+    return isDouble ? (PatchNanFunc)GET_OPTIMIZED(patchNaNs_64f)
+                    : (PatchNanFunc)GET_OPTIMIZED(patchNaNs_32f);
+}
+
+////// finiteMask //////
+
+#if (CV_SIMD || CV_SIMD_SCALABLE)
+
+//TODO: make true SIMD code instead for the rest
+template <typename _Tp, int cn>
+int finiteMaskSIMD_(const _Tp *src, uchar *dst, size_t total)
+{
+    const int osize = 8;
+    int i = 0;
+    for (; i <= (int)total - osize; i += osize)
+    {
+        for (int j = 0; j < osize; j++)
+        {
+            bool finite = true;
+            for (int c = 0; c < cn; c++)
+            {
+                _Tp val = src[i * cn + j * cn + c];
+                finite = finite && !cvIsNaN(val) && !cvIsInf(val);
+            }
+            dst[i + j] = finite ? 255 : 0;
+        }
+    }
+
+    return i;
+}
+
+
+template <>
+int finiteMaskSIMD_<float, 1>(const float *src, uchar *dst, size_t total)
+{
+    const int osize = VTraits<v_uint8>::vlanes();
+    int i = 0;
+    for(; i <= (int)total - osize; i += osize )
+    {
+        v_uint32 vmaskPos = vx_setall_u32(0x7fffffff);
+        v_uint32 vmaskExp = vx_setall_u32(0x7f800000);
+        v_uint32 vv[4];
```

Comment:
Arrays of register won't work with Scalable RVV. It should be replaced with 4 variables: `vv0`, `vv1`, `vv2`, `vv3`.

Reply:
Fixed